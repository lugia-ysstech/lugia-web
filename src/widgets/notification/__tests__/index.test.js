/**
 *
 * create by guorg
 *
 * @flow
 *
 */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import chai from 'chai';
import Adapter from 'enzyme-adapter-react-16';
import { delay } from '@lugia/react-test-utils';

import NotificationDemo from '../demo';
import Notification from '../notification';

const { expect: exp } = chai;
Enzyme.configure({ adapter: new Adapter() });

describe('Notification', () => {
  const getState = (target: any): Object => {
    const state = target
      .children()
      .at(0)
      .instance().state;
    return state;
  };
  it('css', () => {
    const Target = <NotificationDemo />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
  it('Notification state', async () => {
    const target = mount(
      <Notification
        create={false}
        duration={4.5}
        title="今天天气很好！"
        description="因为今天的太阳很大。"
      />
    );
    const state = getState(target);
    expect(state.opening).toBe(true);
    await delay(1000);
    expect(getState(target).opening).toBe(true);
    await delay(3500);
    expect(getState(target).closing).toBe(true);
    await delay(400);
    expect(getState(target).closing).toBe(false);
    expect(getState(target).visible).toBe(false);
  });
});
