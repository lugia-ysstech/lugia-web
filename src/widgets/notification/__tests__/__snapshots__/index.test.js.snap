// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification css 1`] = `
.c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c3 {
  font-size: 1.4rem;
  box-sizing: border-box;
  padding: 2rem 4.4rem;
  background: #fff;
  line-height: 1.5;
  position: relative;
  margin-bottom: 1.6rem;
  box-shadow: 0rem 0rem 0.6rem 0rem #d0d3da;
  border-radius: 0.8rem;
  -webkit-animation: iBnjCl 0.3s ease-in;
  animation: iBnjCl 0.3s ease-in;
  z-index: 99999;
}

.c6 {
  box-sizing: border-box;
}

.c7 {
  font-weight: 500;
  margin-bottom: 0.8rem;
  line-height: 2.4rem;
  display: inline-block;
  word-break: break-all;
  font-size: 1.8rem;
  color: #363B40;
}

.c8 {
  font-weight: 500;
  word-break: break-all;
  font-size: 1.4rem;
  color: #646A6E;
}

.c10 {
  font-size: 1.2rem;
}

.c9 {
  position: absolute;
  right: 1.6rem;
  top: 1.15rem;
  outline: none;
  cursor: pointer;
  color: #9EA4AE;
}

.c9:hover {
  color: #646A6E;
}

.c4 {
  position: absolute;
  line-height: 2.2rem;
  top: 2.2rem;
  left: 1.6rem;
  cursor: default;
  font-size: 1.8rem;
  color: #4d63ff;
}

.c0 {
  display: inline-block;
  margin-bottom: 0;
  box-sizing: border-box;
  touch-action: manipulation;
  cursor: pointer;
  white-space: nowrap;
  font-family: 'Trebuchet MS',Arial,Helvetica,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  text-transform: none;
  outline: 0;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  background-color: white;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #d0d3da;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #d0d3da;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #d0d3da;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #d0d3da;
}

.c2 {
  font-size: 1.4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div>
  <br />
  <br />
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        基本用法
      </span>
    </span>
  </button>
  <br />
  <br />
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        自定义延时关闭
      </span>
    </span>
  </button>
    
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        自定义延时关闭
      </span>
    </span>
  </button>
  <br />
  <br />
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        info
      </span>
    </span>
  </button>
    
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        success
      </span>
    </span>
  </button>
    
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        error
      </span>
    </span>
  </button>
    
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        warning
      </span>
    </span>
  </button>
    
  <br />
  <br />
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        自定义图标
      </span>
    </span>
  </button>
  <br />
  <br />
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        自定义方向-bottomLeft
      </span>
    </span>
  </button>
    
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        自定义方向-bottomRight
      </span>
    </span>
  </button>
    
  <button
    className="c0"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#d0d3da",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "#d0d3da",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "#d0d3da",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "#d0d3da",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
  >
    <span
      className="c1"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#363B40",
            "fontSize": "1.4rem",
          }
        }
      >
        自定义方向-topLeft
      </span>
    </span>
  </button>
  <br />
  <br />
  <div
    className="c3"
    style={
      Object {
        "backgroundColor": "#ccc",
        "borderBottomColor": "blue",
        "borderBottomLeftRadius": "1rem",
        "borderBottomRightRadius": "1rem",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.1rem",
        "borderLeftColor": "blue",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0.1rem",
        "borderRightColor": "blue",
        "borderRightStyle": "solid",
        "borderRightWidth": "0.1rem",
        "borderTopColor": "blue",
        "borderTopLeftRadius": "1rem",
        "borderTopRightRadius": "1rem",
        "borderTopStyle": "solid",
        "borderTopWidth": "0.1rem",
        "boxShadow": "0px 0px 8px 0px red",
        "height": "10rem",
        "marginBottom": "0rem",
        "marginLeft": "6rem",
        "marginRight": "0rem",
        "marginTop": "0rem",
        "width": "50rem",
      }
    }
  >
    <i
      className="lugia-icon-logo_ysstech c4  c5"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "color": "yellow",
        }
      }
    />
    <div
      className="c6"
    >
      <div
        className="c7"
        style={
          Object {
            "color": "red",
          }
        }
      >
        我是标题
      </div>
      <div
        className="c8"
        style={
          Object {
            "color": "yellow",
          }
        }
      >
        <div>
          我是内容
        </div>
      </div>
    </div>
    <div
      className="c9"
      onClick={[Function]}
    >
      <i
        className="lugia-icon-reminder_close_circle_o c10  c5"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "color": "yellow",
          }
        }
      />
    </div>
  </div>
</div>
`;
